NConfiguration.Tests/Serialization/TimeSpanParserTests.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System;
+using System.Globalization;
+using NConfiguration.Serialization.SimpleTypes.Parsing;
+using NUnit.Framework;
+
+namespace NConfiguration.Serialization
+{
+    [TestFixture]
+    public class TimeSpanParserTests
+    {
+        private TimeSpanParser _parser;
+        private CultureInfo _ci;
+
+        static readonly object[] TimeSpanTestCases =
+        {
+            new object[] { "", TimeSpan.Zero},
+            new object[] { "-3d", TimeSpan.FromDays(-3)},
+            new object[] { "3.4d", TimeSpan.FromDays(3.4)},
+            new object[] { "3h30m", TimeSpan.FromHours(3.5) },
+            new object[] { "24h5s", TimeSpan.FromHours(24).Add(TimeSpan.FromSeconds(5))},
+            new object[] { "-72.5d8s", TimeSpan.FromDays(-72.5).Add(TimeSpan.FromSeconds(-8))},
+            new object[] { "-72.5d30.4h500m8s", TimeSpan.FromDays(-72.5)
+                .Add(TimeSpan.FromHours(-30.4))
+                .Add(TimeSpan.FromMinutes(-500))
+                .Add(TimeSpan.FromSeconds(-8))},
+        };
+
+        private static readonly object[] TimeSpanCaseInsensitiveTestCases =
+        {
+            new object[] { "2D3H4M5S", TimeSpan.FromDays(2)
+                .Add(TimeSpan.FromHours(3))
+                .Add(TimeSpan.FromMinutes(4))
+                .Add(TimeSpan.FromSeconds(5))
+            },
+            new object[] { "2D3h4M5s", TimeSpan.FromDays(2)
+                .Add(TimeSpan.FromHours(3))
+                .Add(TimeSpan.FromMinutes(4))
+                .Add(TimeSpan.FromSeconds(5))
+            }
+        };
+
+        [SetUp]
+        public void SetUp()
+        {
+            _ci = CultureInfo.InvariantCulture;
```

Comment:
add tests on the locale in which the comma is used as the fraction separator
for example ru-RU